fga-eps-mds/2020.1-stay-safe-secretary-service
# Refactor DF and SP spiders to catch monthly data

## Issue
Resolves #29 


## Description


### Web Crawler
Replaced DfItem and SpItem to a single CrimesItem with the default fields that DF, SP and all future spiders must have to work with the process_item on pipeline.

Replaced DfPipeline and SpPipeline to a single CrimesPipeline that work with all spiders and process (save in database) each month data (CrimesItem) at a time.

DF and SP spiders refactored according with the changes above and to catch monthly data and save following the JSON model  at the architecture document.. The calls to functions that treat data are now made on the spiders to pass only the formulated monthly data (CrimesItem) to the pipeline.

Added more comments to make the whole process more understandable.

### API
Updated controller and tests according to the changes in the JSON data.

Deleted all .gitkeep files.

Added documentation for crimes endpoints in swagger.

## Checklist:



## Known problems (if appropriate):


The time.sleep() that selenium needs to not try to catch a data while the page is reloading doesn't work if the page reload slower than the normal. A future improvement is make the WebDriverWait works.

## How to validate?

Check if the changes to a single item and pipeline really makes the crawlers easily scalable to more secretaries and if the refactoring in the spiders to catch a monthly data is working fine.

Run the spiders and check if the data saved in the database follows the model at the architecture document.

Check if the crime endpoint in swagger is correct.
